### PR TITLE
Review safemath & safecast

### LINKED ISSUE #682: [ETHEREUM-CONTRACTS] SafeMaths/SafeCast for int96/uint128 types
Currently int96 is used for FlowRate type, and uint128 is used for both IndexValue and IndexUnit types.

Common mistakes can be made with regards to SafeMath & SafeCast, some internal libraries are used in code base to make them secure. These internal libraries could be made more complete, and SuperApp builders can start to use them too.

## Notes

These libraries are only necessary for people using 0.7.x library. For 0.8.x users they should not include them.

- Current internal libraries:
  - packages/ethereum-contracts/contracts/libs/Int96SafeMath.sol
  - packages/ethereum-contracts/contracts/libs/UInt128SafeMath.sol

### PR DESCRIPTION
**Pull request by hellwolf**
- Abandon the idea of keeping special revert messages for the int96/uint128 safemath
- Added testFlowDataEncoding & testFlowOperatorDataEncoding to ConstantFlowAgreementV1Properties testing.

### CODE REVIEW

File: packages/ethereum-contracts/test/foundry/agreements/ConstantFlowAgreementV1.prop.sol [hunk 1]
```diff
@@ -68,4 +92,36 @@ contract ConstantFlowAgreementV1Properties is Test {
...
+        ConstantFlowAgreementV1.FlowData memory b;
+        (exist, b) = cfa.decodeFlowData(wordA);
+        assert(a.timestamp == b.timestamp);
+        assert(a.flowRate == b.flowRate);
+        assert(a.deposit == b.deposit);
+        assert(a.owedDeposit == b.owedDeposit);
+    }
+
+    function testFlowOperatorDataEncoding(uint8 permissions, int96 flowRateAllowance) public {
+        vm.assume(flowRateAllowance > 0);
```
Comment (d10r) [inline packages/ethereum-contracts/test/foundry/agreements/ConstantFlowAgreementV1.prop.sol]: in `_encodeFlowOperatorData`, it checks for >= 0

**Review by d10r — APPROVED** [review]
image

very gud!

File: packages/ethereum-contracts/test/foundry/agreements/ConstantFlowAgreementV1.prop.sol (hunk 1 above)
Comment (hellwolf) [reply packages/ethereum-contracts/test/foundry/agreements/ConstantFlowAgreementV1.prop.sol]: I will make a quick change

Comment (hellwolf) [reply packages/ethereum-contracts/test/foundry/agreements/ConstantFlowAgreementV1.prop.sol]: resolved

**Pull request merged**